src/lib.rs

```diff
@@ -385,36 +416,22 @@ fn handle_packet<H, W>(data: &[u8],
             Command::Kill(Some(_)) => Response::Error(NOT_IMPLEMENTED),
             Command::Reset => Response::Empty,
             Command::ReadRegister(regno) => {
-                match handler.read_register(regno) {
-                    Result::Ok(bytes) => Response::Bytes(bytes),
-                    Result::Err(SimpleError::Error(val)) => Response::Error(val),
-                }
+                Response::from(handler.read_register(regno))
```

Comment:
Because of the symmetric impl with `Into` the conventional way to write this is just `handler.read_register(regno).into()`